bin/ensure-node-modules-present.js

```diff
@@ -0,0 +1,13 @@
+const spawnSync = require( 'child_process' ).spawnSync;
+const fs = require( 'fs' );
+
+if ( ! fs.existsSync( 'node_modules' ) ) {
```

Comment:
> Oh, another thing is: I may have missed some conversations, but I don't understand why the install check is done differently in this hooks case.

[Here's the Slack conversation](https://a8c.slack.com/archives/C03DY3RJ5/p1498158554778108). Basically, for running things like `npm start` or `npm build`, it's important that the dependencies installed are *exactly* what it's specified in `npm-shrinkwrap.json`. But for just running `eslint` over the edited files, the only important thing is that `eslint` is installed. Before, committing after switching branches would take minutes because it would run a pointless `npm install`.

> What are your thoughts regarding using install-if-needed in npm commands, but doing it differently here? Would it make sense to put this logic (and make it more explicit) in the package.json itself?

Good idea, I've done that :)